rust/osm2lanes/src/transform/tags_to_lanes/lane.rs

```diff
@@ -0,0 +1,213 @@
+use super::{Infer, Locale, Oneway, RoadMsg, RoadWarnings, TagKey, Tags};
+use crate::transform::tags_to_lanes::bus::LanesBusScheme;
+
+const LANES: TagKey = TagKey::from("lanes");
+pub struct LanesScheme {
+    pub lanes: Infer<usize>,
+    pub forward: Infer<usize>,
+    pub backward: Infer<usize>,
+    pub bothways: Infer<usize>,
+}
+
+impl LanesScheme {
+    /// Parses and validates the `lanes` scheme (which excludes parking lanes, bike lanes, etc.).
+    /// See https://wiki.openstreetmap.org/wiki/Key:lanes.
+    ///
+    /// Validates `lanes[:{forward,both_ways,backward}]=*` and `centre_turn_lane=yes`.
+    pub(super) fn new(
+        tags: &Tags,
+        oneway: Oneway,
+        lanes_bus: &LanesBusScheme,
+        _locale: &Locale,
+        warnings: &mut RoadWarnings,
+    ) -> Self {
+        // The tags for this schema (which we will validate).
+        let tagged_lanes: Option<usize> = tags.get_parsed(&LANES, warnings);
+        let tagged_forward: Option<usize> = tags.get_parsed(&(LANES + "forward"), warnings);
+        let tagged_backward: Option<usize> = tags.get_parsed(&(LANES + "backward"), warnings);
+        let tagged_bothways: Option<usize> = tags.get_parsed(&(LANES + "both_ways"), warnings);
+        let tagged_center_turn_lanes = match tags.is("centre_turn_lane", "yes") {
```

Comment:
Should we emit a deprecation warning? https://wiki.openstreetmap.org/wiki/Key:centre_turn_lane